hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataReader.java

```diff
@@ -516,6 +516,10 @@ public boolean checkAcls(OzoneObj obj, RequestContext context,
   static String getClientAddress() {
     String clientMachine = Server.getRemoteAddress();
     if (clientMachine == null) { //not a RPC client
+      clientMachine =
+          GrpcOzoneManagerMetadataInterceptor.CLIENT_ADDRESS_KEY.get();
```

Comment:
Apart from this change and the `GrpcOzoneManagerMetadataInterceptor`, the rest are cosmetic changes, right? If that's the case, can we adjust the PR title accordingly?

The random selection works with hRPC but not gRPC due to the lack of clientAddress.

Reply:
I wouldn't say so, but as @adoroszlai suggested, I've split this PR into two parts. The GrpcOzoneManagerMetadataInterceptor-related changes are now in https://github.com/apache/ozone/pull/5591